metals/src/main/scala/scala/meta/internal/metals/ConnectionProvider.scala

```diff
@@ -427,7 +432,12 @@ class ConnectionProvider(
 
       compilations.cancel()
       buildTargetClasses.cancel()
-      diagnostics.reset()
+      // if the last build failed, preserve diagnostics for sbt files
+      if (leaveSbtDiagnostics) {
```

Comment:
Isn't the condition the other way around? If leaveSbtDiagnotstics is true we are actually removing them